### PR TITLE
Add rotate-access-key command

### PR DESCRIPTION
**Pull request by loopingz**
To simplify rotating AccessKey to respect the 90 days principal i added a command to rotate the access key.

### CODE REVIEW

**Comment by JordonPhillips** [comment]
Thanks for the contribution, I really like the idea! I'll have a look when I get some time.

**Comment by loopingz** [comment]
Thanks i really hope it will make it to the next release, I'm trying to push the 90 days expiration on IAM Keys but without the tooling it will be harder to push

**Comment by loopingz** [comment]
Any updates on this one ?

**Comment by henvic** [comment]
Hi. Any news on this one?

**Comment by rdp-cloud** [comment]
One point I would raise.. upon renewal, it would be preferred to 'disable' the previous access keys rather than remove them altogether. Removing them could have disasterous consequences if key management is less than ideal within an organisation.

Naturally, this would affect the renewal workflow as there are only 2 sets of keys available at any given time.

1. search for 'disabled' keys
2. remove any disabled keys
3. generate new keys 
4. set original 'active' key to disabled

**Comment by loopingz** [comment]
@rdp-cloud The problem with disabling is that then it is in the way for the next rotation. With removing if i want to rotate every hour i can.

But I can add an option to not remove and disable instead

File: awscli/examples/configure/rotate-access-keys/_examples.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    aws_access_key_id = renew_access_key
+    aws_secret_access_key = renew_secret_key
+
+    [other]
+    aws_access_key_id = other_access_key
+    aws_secret_access_key = other_secret_key
+
+Then the following commands::
+
+    $ aws --profile=test configure rotate-access-key
```
Comment (tingletech) [inline awscli/examples/configure/rotate-access-keys/_examples.rst]: `--profile=other`?

Comment (loopingz) [reply awscli/examples/configure/rotate-access-keys/_examples.rst]: Correct i ll fix this one

**Comment by mmraz** [comment]
Is this still being considered?  It aggregates a set of steps we currently have to take sequenced by an external script.  Would really be nice to consolidate them down into a single call with a single return code to determine success or failure as an atomic operation.

**Comment by loopingz** [comment]
@mmraz while waiting for AWS team (1 year and a half now) i did this small project: https://github.com/loopingz/yaws it has the two functions i added to awscli but never merged

**Comment by loopingz** [comment]
Why this still has no review or reason, even while reaching to our enterprise support to get things move forward...

**Comment by shaunhickson** [comment]
Thank you for this contribution. I’m very sorry that it took us so long to respond. As key rotation is something that has the potential to impact a number of areas across the AWS ecosystem, we are not able to accept this PR at this time. We are going to have an internal discussion regarding this and once we’ve identified what the next steps will be we will be in touch with more details. We will follow up with you as soon as we can.

**Comment by mmraz** [comment]
> We are going to have an internal discussion regarding this and once we’ve identified what the next steps will be we will be in touch with more details

Can this feature be added to the aws-cli roadmap?  I check there regularly.

**Comment by loopingz** [comment]
I do not understand how key rotation is problematic, it is even a part of white paper, trying to help users to achieve better security does not seem  problematic to me, and as it is just a command if you don't run it it does not rotate any keys by itself.

**Comment by andymac4182** [comment]
@shaunhickson Any update on this? We are using https://github.com/rhyeal/aws-rotate-iam-keys to handle this for our local machines but would be great if it could be built in. We have a forced 90 day expiry on our personal access keys so it would be a lot of help to have this added in natively

**Pull request closed without merging**